Assets/SEE/XR/RadialSelection.cs

```diff
@@ -0,0 +1,409 @@
+    /// <summary>
+    /// The angle between the radial-parts.
+    /// </summary>
+    public float angleBetweenPart = 10;
+    /// <summary>
+    /// The radial-parts which got spawned.
+    /// </summary>
+    private List<GameObject> spawnedParts = new List<GameObject>();
+    /// <summary>
+    /// The transform of the controller.
+    /// </summary>
+    public Transform handTransform;
+    /// <summary>
+    /// The currently selected radial-part.
+    /// </summary>
+    private int currentSelectedRadialPart = -1;
+    /// <summary>
+    /// The selected radial-part.
+    /// </summary>
+    public UnityEvent<int> OnPartSelected;
+    /// <summary>
+    /// The button, which triggers the radial-menu.
+    /// </summary>
+    public InputActionReference radialMenu;
+    /// <summary>
+    /// All actions, which are currently available.
+    /// </summary>
+    List<string> actions = new();
+    /// <summary>
+    /// Alle submenus and their entries.
+    /// </summary>
+    List<(string, List<string>)> subMenus = new List<(string, List<string>)>();
+    /// <summary>
+    /// All entry of a submenu.
+    /// </summary>
+    List<string> menuEntrys = new();
+    /// <summary>
+    /// This is used for the rotate-action, because in this case we
+    /// have a dial, which is getting spawned, and should be despawned, when the action changes.
+    /// </summary>
+    GameObject actionObject;
+    /// <summary>
+    /// The position of the submenu.
+    /// It should be the same, as the position from the mainmenu.
+    /// </summary>
+    Vector3? subMenuPosition;
+    /// <summary>
+    /// The rotation of the submenu.
+    /// It should be the same, as the rotation from the mainmenu.
+    /// </summary>
+    Quaternion? subMenuRotation;
+    /// <summary>
+    /// The hide-action is a special action, because it has sub-actions, which we need to access.
+    /// </summary>
+    public static HideModeSelector HideMode { get; set; }
+    // Awake is always called before any Start functions.
+    private void Awake()
+    {
+        radialMenu.action.performed += RadialMenu;
+        ActionStateTypes.AllRootTypes.PreorderTraverse(Visit);
+        bool Visit(AbstractActionStateType child, AbstractActionStateType parent)
+        {
+            string name;
+
+            if (child is ActionStateType actionStateType)
+            {
+                name = child.Name;
+            }
+            else if (child is ActionStateTypeGroup actionStateTypeGroup)
+            {
+                name = child.Name;
+            }
+            else
+            {
+                throw new System.NotImplementedException($"{nameof(child)} not handled.");
+            }
+
+            // If child is not a root (i.e., has a parent), we will add the entry to
+            // the InnerEntries of the NestedMenuEntry corresponding to the parent.
+            // We know that such a NestedMenuEntry must exist, because we are doing a
+            // preorder traversal.
```

Comment:
The terminology doesn't seem to completely match the rest of the file here (e.g., `NestedMenuEntry`).